### DISCUSSION TITLE
timers and prescalers  - dependencies

### DISCUSSION THREAD

**Discussion by inductosyn** [dbuezas/lgt8fx#314] [General]
First, i am very sorry that i am completely stupid. 
I read LGT English manual, read Atmel manual and can not figure out where is pointed that if timer runs by system clock tacts frequency divided according to CLKPR firstly (at PS0-PS3) and than divided again according to TC* prescaler (in CS*0-CS*2 at TCCR*B) and result uses for timer increment/decrement?

Page 129 says "Meanwhile TC1 can use 10-bit frequency prescaler alone or share it with TC0.
Prescaler divides frequency at system clock clkio or high speed clock rcm2x (internal 32M RC oscillator
output 2 times frequency of rc32m clock)to generate counter clock CLKt1."

Within experiment i understood that clock source divided according to CLKPR  firstly and then according to TCCR*B but where can i get a theoretical basis?

**Comment by dbuezas** [comment]
Not only that, but the lgt328 has a clock doubler for timers 0 and 1

If you are working with timers, this tool may give you a better intuitive understanding of how to do what and how each flag affects behaviour:
https://dbuezas.github.io/arduino-web-timers/#mcu=LGT8F328P&timer=1&clockDoubler=on

If you select "with internals" in the dropdown on the top right, you can toggle each individual bit and see its effects

**Comment by dbuezas** [comment]
Regarding your specific question, the timers are fed from the prescaled cpu clock (via CLKPR). 
In the following diagram from the datasheet you can see that the input clock is `clkio` defined as _"The system clock prescaler"_
![image](https://github.com/dbuezas/lgt8fx/assets/777196/8969faca-8c7c-4ace-94a2-da172b55b37a)

**Comment by dbuezas** [comment]
By the way, there's nothing stupid in your question :)
